### ISSUE TITLE
Logger connector improvements tracker

### ISSUE THREAD

**Issue by carmocca** [PyTorchLightning/pytorch-lightning#7898]
## 🚀 Feature

- [ ] Logger connector read-only properties: https://github.com/PyTorchLightning/pytorch-lightning/pull/7882/files#r647359370
- [x] test `fx_validator` for all functions: https://github.com/PyTorchLightning/pytorch-lightning/blob/764d2c775e2f3568975a60e94520bca8a42b2490/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py#L83. #8498
- [ ] Refactor logging tests - they are currently scattered over several files.
- [ ] Rely on `Metric` to sync. We gather+reduce together (per training type) whereas `torchmetrics` does it separately. Would also need to inject `is_tpu_distributed()` logic. https://github.com/PyTorchLightning/pytorch-lightning/blob/764d2c775e2f3568975a60e94520bca8a42b2490/pytorch_lightning/trainer/connectors/logger_connector/result_new.py#L176-L193

**Fault Tolerant Logging**
- [x] Serialization: https://github.com/PyTorchLightning/pytorch-lightning/pull/7891/commits/bab742e45a3b8c3decb22cc80a64bc01ef57e794 https://github.com/PyTorchLightning/pytorch-lightning/pull/7631/commits/37e75897ac4f16d32a53b603333c8e4360257a1b
- [x] Add `load_from_state_dict` to ResultCollection
- [x] Design loop state save/load hierarchy. WIP in https://github.com/PyTorchLightning/pytorch-lightning/pull/7966/commits/189f0ad152cb8bb94e7c64a920ced512861fd0ec. Intended structure:

```
trainer
    loops:
        fit_loop:
            epoch_loop:
                batch_loop:
                validation_loop:
                    ...
        validate_loop:
            ...
        test_loop:
            ...
------
and each loop has:
a_loop:
    progress
    results
```
- [x] Include the loops state in the checkpoint
